### DISCUSSION TITLE
Database Policies: auth.uid() doesn't work

### DISCUSSION THREAD

**Discussion by madebyfabian** [supabase/supabase#477] [Questions] [answered]
Hey there, first of all, this product is really awesome! Love it so far, even though I discovered a few bugs/strange behaviours.
But one of it, I absolutely can't get it to work:

I have the following table "recipes":

> id: (uuid)
> author_id: (uuid)
> title: (text)
> ...

As you can guess, the "author_id" is the UUID of a user account that owns the recipe and it's the only one who should have access to. Now, when I add a rule like it's in the docs:
```sql
create policy "Individuals can view their own recipes." 
on recipes for
select using (auth.uid() = author_id);
```
the rule is getting applied and the dashboard looks like this: 
![image](https://user-images.githubusercontent.com/32813692/103641883-b0f9b700-4f52-11eb-915b-f437c4ac912c.png)
One strange thing here I notice is that the "auth." is getting stripped off automatically, but maybe that's by purpose? 

So great, by far. But now, when I want to access the table via the JS API:
```js
let { data: recipes } = await supabase
  .from('recipes')
  .select('*')
```
I still get every recipe in the whole database :(, even if the "author_id" field is another ID than the one I am authorized with on the JS client.

When I try to change the policy to just "false" or something else that returns false all the time, than of course I get an empty array when I am asking for recipes in the JS client. 

I tried it with either the SQL command or the creation of policies via UI.

Is this a known issue?

**Comment by alfredosalzillo** [comment]
Have you enabled row-level security ?
```sql
alter table public.recipes
    enable row level security;
```

**Reply by madebyfabian** [reply]
Hello @alfredosalzillo , yep, it’s activated. I tried everything but still can’t get it to work.

**Comment by kiwicopple** [comment, marked as answer]
Hye @madebyfabian - sorry I'm having a hard time reproducing this. Do you have a database dump that I can check? (you can download it from the Dashboard). Make sure to clean anything sensitive before adding it here!

**Reply by madebyfabian** [reply]
Hey @kiwicopple , thanks for reaching out on this. I cannot find any download for the database dump, where on the dashboard exactly can I find it?

**Reply by kiwicopple** [reply]
In DB > Backups

<img width="905" alt="image" src="https://user-images.githubusercontent.com/10214025/104015664-ea9b1e00-51ef-11eb-84b7-6e1813c6de61.png">

**Reply by kiwicopple** [reply]
Hi @madebyfabian - I got the file you added here but I deleted your comment - it contained sensitive info ...

**Reply by madebyfabian** [reply]
Hey @kiwicopple thanks for that, I just didn’t stripped anything out because it‘s only demo-data.

**Reply by kiwicopple** [reply]
Hi @madebyfabian I see in your first image RLS was turned off (that's why the padlock is red). In the dump that you sent me the policy has changed from `select` to `all`.

could you please run 

```sql
drop policy "Allow logged-in full access" ON public.recipes;
create policy "Allow logged-in full access" ON public.recipes for select USING ((auth.uid() = author_id));
```

After doing that your users will only get the recipes which belong to them :)

**Reply by madebyfabian** [reply]
@kiwicopple Thank you for your answer, now it works! Yeah it was different since I tried out many things. I don't know what was the reason, I really tried it for hours, couldn't get it to work. But now it works like a charm :) 
Thank you again! Really enjoy this project so far.

**Comment by Sesa1988** [comment]
I created the table and policy via UI and I always get an emtpy list. If my expression is just true I get all my rows.

![image](https://user-images.githubusercontent.com/26410357/150381750-7b02d46a-17f5-4a3d-8961-7187bd34f099.png)

![image](https://user-images.githubusercontent.com/26410357/150381850-da9ea129-9a8b-4465-9ed4-3bc42ac0bf16.png)


`create policy "Read own portfolios" ON public.portfolios for select USING ((auth.uid() = portfolios.user_id));`

```
var response = await supabaseClient.from('portfolios').select().execute();
    if (response.error == null) {
      final results = response.data as List<dynamic>;
      return results.map((e) => Portfolio.fromMap(e)).toList();
    }
    throw Exception("Error while fetching portfolios");
```

I tried as well everything via script:

```
CREATE TABLE portfolios (
  id bigint GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
  name text,
  user_id uuid references users (id)
);

alter table portfolios
  enable row level security;
```

`create policy "Read own portfolios" ON public.portfolios for select USING ((auth.uid() = public.portfolios.user_id));`
